### PR TITLE
Fix sliders being tracked incorrectly with the wrong keypress

### PR DESCRIPTION
**Pull request by nyquillerium**
- Closes https://github.com/ppy/osu/issues/2647

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs [hunk 1]
```diff
@@ -146,24 +147,49 @@ private set
         }
 
         private bool canCurrentlyTrack => Time.Current >= slider.StartTime && Time.Current < slider.EndTime;
+        private OsuAction? trackingAction;
 
         protected override void Update()
         {
             base.Update();
-
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs]: revert

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs [hunk 2]
```diff
@@ -146,24 +147,49 @@ private set
         }
 
         private bool canCurrentlyTrack => Time.Current >= slider.StartTime && Time.Current < slider.EndTime;
+        private OsuAction? trackingAction;
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs]: newline above

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs [hunk 3]
```diff
@@ -146,24 +147,49 @@ private set
...
+
+        public bool OnPressed(OsuAction action)
+        {
+            bool hit = false;
+
+            switch (action)
+            {
+                case OsuAction.LeftButton:
+                case OsuAction.RightButton:
+                    hit = IsHovered;
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs]: pretty sure this conditional is gonna be wrong, based on the conditionals on `IsTracking`.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs (hunk 1 above)
Comment (nyquillerium) [reply osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs]: appveyor's failing now though

**Comment by nyquillerium** [comment]
Actually, I realized there's a little more testing to do, and the conditions could read better so I'm going to close this for now. Will re-open when its more or less ready to merge.

**Pull request closed without merging**